### ISSUE TITLE
fcntl.ioctl() regression: mutate_flag now mandatory for read-only buffer

### ISSUE THREAD

**Issue by julian-klode** [python/typeshed#3881]
#3680 now mandates the presence of a boolean mutate_flag in case of an read-only bytes buffer for the argument, which seems wrong, it should be optional (it does not do anything).

This breaks code like this:

```buf = fcntl.ioctl(fileno, termios.TIOCGWINSZ, 8 * b' ')```

which now pointlessly needs to be written as

```buf = fcntl.ioctl(fileno, termios.TIOCGWINSZ, 8 * b' ', False)```

or reworked to use a bytearray.

**Comment by julian-klode** [comment]
This also breaks compatibility with Python 2 code, as it does not accept the mutate_flag argument.

```
TypeError                                 Traceback (most recent call last)
<ipython-input-6-73e86fea1fbd> in <module>()
----> 1 fcntl.ioctl(2, termios.TIOCGWINSZ, 8 * b' ', True)

TypeError: ioctl requires a file or file descriptor, an integer and optionally an integer or buffer argument
```

**Comment by julian-klode** [comment]
Fix in #3882
